rasa/core/domain.py

```diff
@@ -775,6 +787,58 @@ def intent_config(self, intent_name: Text) -> Dict[Text, Any]:
     def intents(self):
         return sorted(self.intent_properties.keys())
 
+    @staticmethod
+    def get_symmetric_difference(
+        domain_elements: Union[List[Text], Set[Text]],
+        training_data_elements: Optional[Set[Text]],
+    ) -> Dict[Text, Any]:
+        """Get symmetric difference between a set of domain elements and a set of
+        training data elements.
+
+        Returns a dictionary containing a list of items found in the `domain_elements`
+        but not in `training_data_elements` at key `in_domain`, and a list of items
+        found in `training_data_elements` but not in `domain_elements` at key
+        `in_training_data_set`.
+        """
+
+        if not training_data_elements:
+            training_data_elements = set()
+
+        diff_dict = defaultdict(list)
+        for e in set(domain_elements) ^ set(training_data_elements):
+            diff_dict[
```

Comment:
I think this is unecessary complex, I'd rather keep the differences in separate variables and then return them as a dict later, something along these lines:
```python
in_domain_diff = set(domain_elements) - set(training_data_elements)
in_training_data_diff = set(training_data_elements) - set(domain_elements)

return {"in_domain": in_domain_diff, "in_training_data": in_training_data_diff}
```